# Find Pv1 studies 


In [2]:
from malariagen_data.pv4 import Pv4
import pandas as pd
import numpy as np
import allel
import dask.array as da
import collections
import math

In [90]:
metadata_fn = '../supplementary_files/Pv4_populations_full_sample_metadata_20190430.txt'

In [91]:
df_samples_all = pd.read_csv(metadata_fn, sep='\t', index_col=0)

In [7]:
# Are usable study samples the same for both 

In [13]:
useable_studies = [
    '1128-PV-GSKMULTI-RAYNER', '1154-PV-TH-PRICE', '1156-PV-ID-PRICE', '1157-PV-MULTI-PRICE', 
    'X0001-PV-MULTI-HUPALO2016', 'X0002-PV-KH-PAROBEK2016'
]

In [14]:
pv4_metadata.loc[(pv4_metadata['Study'].isin(useable_studies))].shape

(1299, 15)

In [15]:
df_samples_all.loc[(df_samples_all['Study'].isin(useable_studies))].shape

(1656, 70)

In [ ]:
# Are the number of samples in Pv1 studies the same for both

In [ ]:
pv4_metadata.loc[(pv4_metadata['Study'].isin(useable_studies))].shape

In [28]:
pv1_studies = list(df_samples_all.loc[df_samples_all['in_pv_10']].Study.unique())

In [30]:
unique_pv1_studies = []
for study in pv1_studies: 
    if study not in useable_studies: 
        unique_pv1_studies.append(study)

In [59]:
samples1 = list(pv4_metadata.loc[(pv4_metadata['Study'].isin(unique_pv1_studies))].Sample)

In [62]:
samples2 = list(df_samples_all.loc[(df_samples_all['Study'].isin(unique_pv1_studies))].index)

In [63]:
if samples1 != samples2: 
    print('oh oh')

In [35]:
# Are the studies the same for both? 
pv4_metadata.Study.value_counts()

1157-PV-MULTI-PRICE                667
1154-PV-TH-PRICE                   359
1128-PV-MULTI-GSK                  357
X0001-PV-MULTI-HUPALO2016          195
1098-PF-ET-GOLASSA                  88
1044-PF-KH-FAIRHURST                82
X0002-PV-KH-PAROBEK2016             78
X0009-PV-ET-LO                      24
1050-PV-PN-MUELLER                  20
1049-PV-VN-BONI                     13
1046-PV-BR-FERRERIA                  5
1052-PF-TRAC-WHITE                   4
1047-PV-LK-KARUNAWEERA               2
1102-PF-MG-RANDRIANARIVELOJOSIA      1
Name: Study, dtype: int64

In [36]:
df_samples_all.Study.value_counts()

1157-PV-MULTI-PRICE                667
1128-PV-GSKMULTI-RAYNER            357
1156-PV-ID-PRICE                   253
X0001-PV-MULTI-HUPALO2016          195
1154-PV-TH-PRICE                   106
1098-PF-ET-GOLASSA                  88
1044-PF-KH-FAIRHURST                82
X0002-PV-KH-PAROBEK2016             78
X0009-PV-ET-LO                      24
1050-PV-PN-MUELLER                  20
1049-PV-VN-BONI                     13
1046-PV-BR-FERRERIA                  5
1052-PF-TRAC-WHITE                   4
1047-PV-LK-KARUNAWEERA               2
1102-PF-MG-RANDRIANARIVELOJOSIA      1
Name: Study, dtype: int64

In [41]:
#check which studies are not in the other one 
all_samples_studies = list(df_samples_all.Study.unique())
pv4_studies = list(pv4_metadata.Study.unique())

In [49]:
print('Studies in pv4, not in old metadata')
for study in pv4_studies: 
    if study not in all_samples_studies: 
        print(study)

Studies in pv4, not in old metadata
1128-PV-MULTI-GSK


In [50]:
print('Studies in old metadata not in Pv4')
for study in all_samples_studies: 
    if study not in pv4_studies:
        print(study)

Studies in old metadata not in Pv4
1156-PV-ID-PRICE
1128-PV-GSKMULTI-RAYNER


In [47]:
for study in pv4_studies: 
    if study in all_samples_studies: 
        length1 = len(pv4_metadata.loc[pv4_metadata.Study == study])
        length2 = len(df_samples_all.loc[df_samples_all.Study == study])
        if length1 != length2: 
            print(study, length1, length2)

1154-PV-TH-PRICE 359 106


* 1128-PV-MULTI-GSK is 1128-PV-GSKMULTI-RAYNER in Pv4 
* 1154-PV-TH-PRICE has 359 samples in pv4 and 106 in the old metadata 
* 1156-PV-ID-PRICE is in old metadata (253 samples) and not in pv4 

In [48]:
pv4_metadata.Study.unique()

array(['X0009-PV-ET-LO', 'X0001-PV-MULTI-HUPALO2016',
       'X0002-PV-KH-PAROBEK2016', '1154-PV-TH-PRICE',
       '1052-PF-TRAC-WHITE', '1044-PF-KH-FAIRHURST',
       '1157-PV-MULTI-PRICE', '1050-PV-PN-MUELLER', '1128-PV-MULTI-GSK',
       '1046-PV-BR-FERRERIA', '1047-PV-LK-KARUNAWEERA', '1049-PV-VN-BONI',
       '1102-PF-MG-RANDRIANARIVELOJOSIA', '1098-PF-ET-GOLASSA'],
      dtype=object)

Change 1128-PV-GSKMULTI-RAYNER to 1128-PV-MULTI-GSK and remove 1156-PV-ID-PRICE

In [ ]:
# How to run

In [3]:
metadata_fn = '../supplementary_files/Pv4_populations_full_sample_metadata_20190430.txt'
df_samples_all = pd.read_csv(metadata_fn, sep='\t', index_col=0)

In [9]:
pv4 = Pv4("gs://pv4_staging/")
pv4_metadata = pv4.sample_metadata()
pv4_metadata.head()

Sample           Study   Site First-level administrative division  \
0  BBH-1-125  X0009-PV-ET-LO  Jimma                    Ethiopia: Oromia   
1  BBH_1_132  X0009-PV-ET-LO  Jimma                    Ethiopia: Oromia   
2  BBH_1_137  X0009-PV-ET-LO  Jimma                    Ethiopia: Oromia   
3  BBH_1_153  X0009-PV-ET-LO  Jimma                    Ethiopia: Oromia   
4  BBH_1_162  X0009-PV-ET-LO  Jimma                    Ethiopia: Oromia   

    Country       Lat       Long  Year         ENA  \
0  Ethiopia  7.683331  36.851318  2016  ERR2678989   
1  Ethiopia  7.683331  36.851318  2016  ERR2678991   
2  Ethiopia  7.683331  36.851318  2016  ERR2679003   
3  Ethiopia  7.683331  36.851318  2016  ERR2678992   
4  Ethiopia  7.683331  36.851318  2016  ERR2678993   

  All samples same individual Population  % callable  QC pass  \
0                   BBH-1-125         AF       88.52     True   
1                   BBH_1_132         AF       90.20     True   
2                   BBH_1_137         AF       87.09     True   
3                   BBH_1_153         AF       90.60     True   
4                   BBH_1_162         AF       91.67     True   

  Exclusion reason  Is returning traveller  
0     Analysis_set                   False  
1     Analysis_set                   False  
2     Analysis_set                   False  
3     Analysis_set                   False  
4     Analysis_set                   False

In [5]:
useable_studies = [
    '1128-PV-MULTI-GSK', '1154-PV-TH-PRICE', '1157-PV-MULTI-PRICE', 'X0001-PV-MULTI-HUPALO2016', 
    'X0002-PV-KH-PAROBEK2016'
]

In [6]:
pv4_metadata.loc[(pv4_metadata['Study'].isin(useable_studies))].shape

(1656, 15)

In [7]:
df_samples_all = df_samples_all.reset_index()

In [8]:
pv4_metadata = pv4_metadata.merge(df_samples_all[['Sample','in_pv_10']], on='Sample', how='left')

In [9]:
pv4_metadata = pv4_metadata.loc[(pv4_metadata['Study'].isin(useable_studies)) | pv4_metadata.in_pv_10]

In [10]:
pv4_metadata

Sample                      Study               Site  \
5        Brazil01  X0001-PV-MULTI-HUPALO2016         Acrelândia   
6        Brazil02  X0001-PV-MULTI-HUPALO2016  Plácido de Castro   
7        Brazil03  X0001-PV-MULTI-HUPALO2016  Plácido de Castro   
8        Brazil08  X0001-PV-MULTI-HUPALO2016         Acrelândia   
9        Brazil12  X0001-PV-MULTI-HUPALO2016  Plácido de Castro   
...           ...                        ...                ...   
1890  india_C1060  X0001-PV-MULTI-HUPALO2016            Chennai   
1891   malagasy15  X0001-PV-MULTI-HUPALO2016         Madagascar   
1892   malagasy19  X0001-PV-MULTI-HUPALO2016       Ampasimpotsy   
1893    malagasy8  X0001-PV-MULTI-HUPALO2016       Ampasimpotsy   
1894   mauritania  X0001-PV-MULTI-HUPALO2016       Mauritania I   

     First-level administrative division     Country   Lat  Long  Year  \
5                           Brazil: Acre      Brazil -10.0 -67.0  2011   
6                           Brazil: Acre      Brazil -10.2 -67.2  2011   
7                           Brazil: Acre      Brazil -10.2 -67.2  2011   
8                           Brazil: Acre      Brazil -10.0 -67.0  2011   
9                           Brazil: Acre      Brazil -10.2 -67.2  2011   
...                                  ...         ...   ...   ...   ...   
1890                   India: Tamil Nadu       India  13.1  80.2  2013   
1891                          Madagascar  Madagascar -20.0  46.0  2010   
1892            Madagascar: Antananarivo  Madagascar -19.4  47.8  2010   
1893            Madagascar: Antananarivo  Madagascar -19.4  47.8  2010   
1894                          Mauritania  Mauritania  20.0 -11.0  1996   

                                                    ENA  \
5                                 SRR1564575,SRR1568121   
6                                 SRR1564605,SRR1568116   
7                                 SRR1567957,SRR1568197   
8                                 SRR1568176,SRR1568808   
9                                 SRR1564700,SRR1568108   
...                                                 ...   
1890                                         SRR1740119   
1891                                          SRR828416   
1892                                          SRR572651   
1893                                          SRR570031   
1894  SRR332408,SRR332409,SRR332410,SRR332411,SRR332...   

     All samples same individual  Population  % callable  QC pass  \
5                       Brazil01         LAM       50.16     True   
6                       Brazil02         LAM        9.39    False   
7                       Brazil03         LAM        0.47    False   
8                       Brazil08         LAM        5.86    False   
9                       Brazil12         LAM       30.67    False   
...                          ...         ...         ...      ...   
1890                 india_C1060         WAS        0.94    False   
1891                  malagasy15  unassigned       87.04     True   
1892                  malagasy19  unassigned       91.52     True   
1893                   malagasy8  unassigned       90.10     True   
1894                  mauritania  unassigned       89.39     True   

             Exclusion reason  Is returning traveller  in_pv_10  
5                Analysis_set                   False     False  
6     Less_than_50pc_callable                   False     False  
7     Less_than_50pc_callable                   False     False  
8     Less_than_50pc_callable                   False     False  
9     Less_than_50pc_callable                   False     False  
...                       ...                     ...       ...  
1890  Less_than_50pc_callable                   False     False  
1891    Unassigned_population                   False     False  
1892    Unassigned_population                   False     False  
1893    Unassigned_population                   False     False  
1894    Unassigned_population                   False

In [16]:
df_samples_all[['Sample','in_pv_10']].to_csv('../supplementary_files/Samples_in_Pv1.tsv', sep='\t', index=False)

In [5]:
#Final solution 

useable_studies = [
    '1128-PV-MULTI-GSK', '1154-PV-TH-PRICE', '1157-PV-MULTI-PRICE', 'X0001-PV-MULTI-HUPALO2016', 
    'X0002-PV-KH-PAROBEK2016'
]

pv1_samples = pd.read_csv('../supplementary_files/Samples_in_Pv1.tsv', sep='\t')
pv4_metadata = pv4_metadata.merge(pv1_samples, on='Sample', how='left')

In [7]:
pv4_metadata = pv4_metadata.loc[(pv4_metadata['Study'].isin(useable_studies)) | pv4_metadata.in_pv_10]

In [8]:
pv4_metadata

Sample                      Study               Site  \
5        Brazil01  X0001-PV-MULTI-HUPALO2016         Acrelândia   
6        Brazil02  X0001-PV-MULTI-HUPALO2016  Plácido de Castro   
7        Brazil03  X0001-PV-MULTI-HUPALO2016  Plácido de Castro   
8        Brazil08  X0001-PV-MULTI-HUPALO2016         Acrelândia   
9        Brazil12  X0001-PV-MULTI-HUPALO2016  Plácido de Castro   
...           ...                        ...                ...   
1890  india_C1060  X0001-PV-MULTI-HUPALO2016            Chennai   
1891   malagasy15  X0001-PV-MULTI-HUPALO2016         Madagascar   
1892   malagasy19  X0001-PV-MULTI-HUPALO2016       Ampasimpotsy   
1893    malagasy8  X0001-PV-MULTI-HUPALO2016       Ampasimpotsy   
1894   mauritania  X0001-PV-MULTI-HUPALO2016       Mauritania I   

     First-level administrative division     Country   Lat  Long  Year  \
5                           Brazil: Acre      Brazil -10.0 -67.0  2011   
6                           Brazil: Acre      Brazil -10.2 -67.2  2011   
7                           Brazil: Acre      Brazil -10.2 -67.2  2011   
8                           Brazil: Acre      Brazil -10.0 -67.0  2011   
9                           Brazil: Acre      Brazil -10.2 -67.2  2011   
...                                  ...         ...   ...   ...   ...   
1890                   India: Tamil Nadu       India  13.1  80.2  2013   
1891                          Madagascar  Madagascar -20.0  46.0  2010   
1892            Madagascar: Antananarivo  Madagascar -19.4  47.8  2010   
1893            Madagascar: Antananarivo  Madagascar -19.4  47.8  2010   
1894                          Mauritania  Mauritania  20.0 -11.0  1996   

                                                    ENA  \
5                                 SRR1564575,SRR1568121   
6                                 SRR1564605,SRR1568116   
7                                 SRR1567957,SRR1568197   
8                                 SRR1568176,SRR1568808   
9                                 SRR1564700,SRR1568108   
...                                                 ...   
1890                                         SRR1740119   
1891                                          SRR828416   
1892                                          SRR572651   
1893                                          SRR570031   
1894  SRR332408,SRR332409,SRR332410,SRR332411,SRR332...   

     All samples same individual  Population  % callable  QC pass  \
5                       Brazil01         LAM       50.16     True   
6                       Brazil02         LAM        9.39    False   
7                       Brazil03         LAM        0.47    False   
8                       Brazil08         LAM        5.86    False   
9                       Brazil12         LAM       30.67    False   
...                          ...         ...         ...      ...   
1890                 india_C1060         WAS        0.94    False   
1891                  malagasy15  unassigned       87.04     True   
1892                  malagasy19  unassigned       91.52     True   
1893                   malagasy8  unassigned       90.10     True   
1894                  mauritania  unassigned       89.39     True   

             Exclusion reason  Is returning traveller  in_pv_10  
5                Analysis_set                   False     False  
6     Less_than_50pc_callable                   False     False  
7     Less_than_50pc_callable                   False     False  
8     Less_than_50pc_callable                   False     False  
9     Less_than_50pc_callable                   False     False  
...                       ...                     ...       ...  
1890  Less_than_50pc_callable                   False     False  
1891    Unassigned_population                   False     False  
1892    Unassigned_population                   False     False  
1893    Unassigned_population                   False     False  
1894    Unassigned_population                   False